src/client/src/logger.ts

```diff
@@ -0,0 +1,48 @@
+import logdown from 'logdown'
+
+enum DebugLevel {
+  error = 'error',
+  warn = 'warn',
+  debug = 'debug',
+  info = 'info',
+}
+
+export enum DebugType {
+  Error = 'Error',
+  Warning = 'Warning',
+  Debug = 'Debug',
+  Subscribing = 'Subscribing',
+  Subscribed = 'Subscribed',
+  Received = 'Received',
+  Executing = 'Executing',
+  Execute = 'Execute',
+  Info = 'Info',
+}
+
+const levelColour = {
+  error: 'Crimson',
+  warn: 'Tomato',
+  debug: 'Orange',
+  info: 'RoyalBlue',
+}
+
+export const createLogger = (debugLevel: DebugLevel) => {
+  const colour = Object.keys(levelColour)
+    .filter(key => key === debugLevel)
+    .map(level => levelColour[level])
+    .join()
+
+  return (name: string, debugType?: DebugType) =>
+    debugType !== undefined
+      ? (...args: any[]) => logdown(`app:${name}${debugType}`, { prefixColor: colour })[debugLevel](...args)
+      : (...args: any[]) => logdown(`app:${name}`, { prefixColor: colour })[debugLevel](...args)
```

Comment:
Use `arguments`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments

Example:
```() => logdown(`app:${name}${debugType}`, { prefixColor: colour })[debugLevel].apply(null, arguments)```


Try something like that(let me know if it works or not).

Reply:
https://stackoverflow.com/questions/41731854/why-do-arrow-functions-not-have-the-arguments-array

Reply:
this does not work really well with arrow function, but it is a very good thing to know about 👍

Reply:
Ah you're right, good catch!